applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/HistoryWidget/HistoryWidgetViewComponent.cs

```diff
@@ -0,0 +1,146 @@
+            string? entityTypeFullName = null;
+            CancellationToken cancellationToken = default;
+
+            var entityChanges = await _auditLogRepository.GetEntityChangeListAsync(
+                sorting,
+                maxResultCount,
+                skipCount,
+                auditLogId,
+                startTime, endTime,
+                changeType,
+                entityId,
+                entityTypeFullName,
+                includeDetails,
+                cancellationToken);
+
+            foreach (var entityChange in entityChanges)
+            {
+                foreach (var propertyChange in entityChange.PropertyChanges)
+                {
+                    if (propertyChange.PropertyName == filterPropertyName)
+                    {
+                        string origninalValue = propertyChange.OriginalValue != null ? propertyChange.OriginalValue.Replace("\"", "") : "";
+                        string newValue = propertyChange.NewValue != null ? propertyChange.NewValue.Replace("\"", "") : "";
+                        HistoryDto historyDto = new HistoryDto()
+                        {
+                            OriginalValue = lookupDictionary != null ? lookupDictionary[origninalValue] : origninalValue,
+                            NewValue = lookupDictionary != null ? lookupDictionary[newValue] : newValue,
+                            ChangeTime = entityChange.ChangeTime,
+                            UserName = await LookupUserName(entityChange.AuditLogId)
+                        };
+                        historyList.Add(historyDto);
+                    }
+                }
+            }
+            return historyList;
+        }
+
+        public async Task<Dictionary<string, string>> GetApplicationStatusDict() {
+            List<ApplicationStatus> applicationStatusList = await _applicationStatusRepository.GetListAsync();
+            Dictionary<string, string> applicationStatusDict = new Dictionary<string, string>();
+            foreach (var applicationStatus in applicationStatusList)
+            {
+                applicationStatusDict.Add(applicationStatus.Id.ToString(), applicationStatus.InternalStatus);
+            }
+            return applicationStatusDict;
+        }
+
+        public async Task<string> LookupUserName(Guid auditLogId)
+        {
+            string userName = "";
+            AuditLog auditLog = await _auditLogRepository.GetAsync(auditLogId);
+            if (auditLog != null && auditLog.UserId != null)
+            {       
+                string? userIdStr = auditLog.UserId.ToString();
+                Guid.TryParse(userIdStr, out Guid userId);
+                var user = await _identityUserAppService.GetAsync(userId);
+                userName = $"{user.Name} {user.Surname}";
+            }
+            
+            return userName;
+        }
+    }
+}
+
+public class HistoryWidgetStyleBundleContributor : BundleContributor
```

Comment:
Move the contributor class to appease SonarQube: https://rules.sonarsource.com/csharp/RSPEC-3903/